AK/Vector.h

```diff
@@ -314,15 +321,16 @@ requires(!IsRvalueReference<T>) class Vector {
         MUST(try_prepend(values, count));
     }
 
-    // FIXME: What about assigning from a vector with lower inline capacity?
-    Vector& operator=(Vector&& other)
+    template<class U, size_t inline_capacity_u>
+    requires CanMoveIn<inline_capacity, inline_capacity_u>
+    Vector& operator=(Vector<U, inline_capacity_u>&& other)
     {
-        if (this != &other) {
+        if (reinterpret_cast<void*>(this) != reinterpret_cast<void*>(&other)) {
```

Comment:
This equality check should only happen if the types are the same, so instead:
```suggestion
        if constexpr (inline_capacity == inline_capacity_u) {
            if (this != &other) {
```

(plus the ensuing indent)

Reply:
Should be resolved now.

Reply:
Hmm, no, the self-assignment check should only happen when the objects are of the same type (my suggestion was wrong, sorry), it's probably better to check for equality and early return if the types are the same.

Reply:
We still need to keep the non-templated `Vector& operator=(Vector&&)` for it to be considered a *move assignment operator* by the language:

https://en.cppreference.com/w/cpp/language/move_assignment
> A move assignment operator is a **non-template** non-static member function with the name `operator=`

We could have a different overload that handles the case with a different inline capacity, see e.g. for `OwnPtr`:
https://github.com/LadybirdBrowser/ladybird/blob/0a3082ef05e8e7a963d3a208acd6efee0ff5c848/AK/OwnPtr.h#L75-L88